src/Pynest2D.sip

```diff
@@ -0,0 +1,153 @@
+        PyObject* iterator = PyObject_GetIter(sipPy);
+        if(!sipIsErr) //We're only checking if it can be converted.
+        {
+            //Must be any iterable.
+            const bool iterable = (iterator != NULL);
+            Py_XDECREF(iterator);
+            return iterable;
+        }
+
+        std::vector<Item*>* storage_vector = new std::vector<Item*>();
+        PyObject* list_item;
+        while(list_item = PyIter_Next(iterator))
+        {
+            if(!sipCanConvertToInstance(list_item, sipClass_Item, SIP_NOT_NONE))
+            {
+                PyErr_Format(PyExc_TypeError, "Object in iterable cannot be converted to a libnest2d Item.");
+                *sipIsErr = 1;
+                break;
+            }
+            int state;
+            Item* item = reinterpret_cast<Item*>(sipConvertToInstance(list_item, sipClass_Item, 0, SIP_NOT_NONE, &state, sipIsErr));
+            if(!*sipIsErr)
+            {
+                storage_vector->push_back(item);
+            }
+
+            sipReleaseInstance(item, sipClass_Item, state);
+            Py_DECREF(list_item);
+        }
+
+        Py_DECREF(iterator);
+        if(*sipIsErr)
+        {
+            delete storage_vector;
+            return 0;
+        }
+        *sipCppPtr = storage_vector;
+        return sipGetState(sipTransferObj);
+    %End
+};
+
+long nest(std::vector<Item*> items, Box bin, long distance = 1, NfpConfig config = NfpConfig());
+%Docstring
+    Arrange a number of items in bins, packing them closely with the goal to
+    minimise the number of bins necessary to pack all given items.
+
+    This function uses the FirstFit selector and NfpPPlacer in its algorithms,
+    and Clipper as geometry back-end.
+
+    The result of the packing is stored in the transformation of the input
+    items. You can get the transformed items by calling transformedShape() on
+    each of the items, or by looking at their rotation and translation. Items
+    may also be placed on alternative bins if the first bin is not big enough to
+    fit all items.
+    :param items: An iterable of Item instances which are the items that need to
+    be placed in the bins.
+    :param bin: The shape in which to place these items.
+    :param distance: (Default 1.) The distance to hold between adjacent items in
+    the final pack result. Items will not get placed closer than this distance.
+    Must be at least 1 unit.
+    :param config: An NfpConfig instance that contains configuration options to
+    specialize how objects get placed. Defaults will be used if this is not
+    provided.
+    :return: The number of bins necessary to fit all items.
+%End
+%MethodCode
+    //Pre-process distance.
+    if(a2 <= 0)
+    {
+        a2 = 1; //Need to keep at least 1 unit of distance, or the nester breaks.
+    }
+
+    //Pre-process items.
+    std::vector<Item> to_arrange;
+    to_arrange.reserve(a0->size());
+    for(Item* item : *a0)
+    {
+        to_arrange.push_back(*item);
```

Comment:
use `emplace_back(...)`

Reply:
Will likely not help in this case (because of compiler-optimisation), and may even hurt (likelihood: very small, but still). See the accepted answer here:  https://stackoverflow.com/questions/10890653/why-would-i-ever-use-push-back-instead-of-emplace-back